# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1464421719 -36000
#      Sat May 28 17:48:39 2016 +1000
# Node ID f382293bc11da3572c323429a52c27aea9ed94b3
# Parent  6b4f6339df911e59ad9fce54ecfd08c993182338
Include @class directories in the output of what() (bug #48035)

* what.m: Check isdir(fullfile(...)), not isdir (n).

diff --git a/scripts/miscellaneous/what.m b/scripts/miscellaneous/what.m
--- a/scripts/miscellaneous/what.m
+++ b/scripts/miscellaneous/what.m
@@ -111,17 +111,17 @@ function retval = what (dir)
       if (strcmp (e, ".m"))
         w.m{end+1} = n;
       elseif (strcmp (e, ".mat"))
         w.mat{end+1} = n;
       elseif (strcmp (e, ".oct"))
         w.oct{end+1} = n;
       elseif (strcmp (e, mexext ()))
         w.mex{end+1} = n;
-      elseif (n(1) == "@" && isdir (n))
+      elseif (n(1) == "@" && isdir (fullfile (dir, n)))
         w.classes{end+1} = n;
       endif
     endif
   endfor
 
   if (nargout == 0)
     __display_filenames__ ("M-files in directory", w.path, w.m);
     __display_filenames__ ("\nMAT-files in directory", w.path, w.mat);
